subsys/bluetooth/common/dummy.c

```diff
@@ -29,12 +29,26 @@ BUILD_ASSERT(CONFIG_BT_HCI_TX_PRIO < CONFIG_BT_RX_PRIO);
 BUILD_ASSERT(CONFIG_BT_CTLR_RX_PRIO < CONFIG_BT_HCI_TX_PRIO);
 #endif /* CONFIG_BT_CTLR */
 
-/* Immediate logging is not supported with the software-based Link Layer
- * since it introduces ISR latency due to outputting log messages with
- * interrupts disabled.
+
+#if defined(CONFIG_LOG_BACKEND_UART) ||                        \
```

Comment:
LOG_BACKEND_NATIVE_POSIX is not needed in this list, because we check CONFIG_ARCH_POSIX in the outer if.

The UART and XTENSA_SIM does not have any irq_lock apart from the IMMEDIATE_CLEAN_OUTPUT, so they could be allowed.
So in my opinion the check should be:
```
#if (defined(CONFIG_LOG_BACKEND_RTT) &&                     \
     (defined(CONFIG_SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL) || \
      defined(CONFIG_LOG_BACKEND_RTT_MODE_BLOCK))) ||       \
    defined(CONFIG_LOG_BACKEND_NET) ||                      \
    defined(CONFIG_LOG_IMMEDIATE_CLEAN_OUTPUT)
```
This also has the correct alignment.

Reply:
I thought the indentation must be made of tab and not spaces ?

Reply:
Depends on where you want to align. A tab is equal to 8 spaces.
If you align starting from the beginning of the line then you using tabs first, then spaces.
So to align on 5, it is 5 spaces, if you align on 19, then it is 2 tabs + 3 spacess.
If you align not from the beginning of the line it is spaces only.

```
/* Only spaces here */
#define BT_GATT_CHRC_BROADCAST			0x01
#define BT_GATT_CHRC_READ			0x02

/* Tabs + spaces */
	BT_GATT_CHARACTERISTIC(BT_UUID_CENTRAL_ADDR_RES,
			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ, /* 3 tabs + 7 spaces */
			       read_central_addr_res, NULL, NULL),
```